[integ-tests] Run test_efa with p6-b200

hanwen-cluster opened the pull request:
### Description of changes
[integ-tests] Refactor get_ultraserver_capacity_reservation_id to be more generic considering instance type]
[integ-tests] Add p6-b200 to test_efa
See commits descriptions for details

### Tests
* test_efa with p6-b200 has passed on Ubuntu22

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

hanwen-cluster committed:
[integ-tests] Refactor get_ultraserver_capacity_reservation_id to be more generic considering instance type

1. Rename the function to get_capacity_reservation_id
2. In the function, use describe_capacity_reservations instead of describe_capacity_block_status because describe_capacity_reservations includes information about instance type
3. Only retrieve reservation with the right instance type
4. Let the caller of the function determines how many instances are needed

hanwen-cluster committed:
[integ-tests] Add p6-b200 to test_efa

p6-b200 requires a capacity block. This commit makes the test to automatically find the capacity block by reusing the same logic as test_gb200. This requires a little bit of refactor

Using capacity block requires using RunInstances instead of CreateFleet. Therefore, this commit changes the cluster configuration file of test_efa to always use RunInstances

himani2411 commented on tests/integration-tests/tests/common/data/mpi/mpi_submit_openmpi.sh:
```diff
@@ -3,4 +3,4 @@ set -e
 rm -f /shared/mpi.out
 module load openmpi
-mpirun --map-by ppr:1:node --timeout 20 "ring" >> /shared/mpi.out
+mpirun --map-by ppr:1:node --timeout 60 "ring" >> /shared/mpi.out
```
why are we increasing? When I tested on gb200 it would complete in 7 seconds

hanwen-cluster replied:
MPI jobs intermittently takes longer and time out on p6-b200.48xlarge.
I removed this increase of timeout and skipped p6-b200 with MPI jobs as discussed

himani2411 commented on tests/integration-tests/tests/common/nccl_common.py:
```diff
@@ -69,6 +69,7 @@ def install_and_run_nccl_benchmarks(remote_command_executor, mpi_module, schedul
         "p4d.24xlarge": 26.0,
         # p5.48xlarge - Expected "in-place busbw" bandwidth with 2 nodes, 8 tasks per node is about 250GB/s
         "p5.48xlarge": 250.0,
+        "p6-b200.48xlarge": 300,
```
where did we get this nos from?

himani2411 replied:
we should chnage the nos for gb200 too

hanwen-cluster replied:
This is an arbitrary number. Without the number, the test fails because of key doesn't exist in the dictionary

hanwen-cluster committed:
[integ-tests] Add p6-b200 NCCL baseline

hanwen-cluster committed:
[integ-tests] Temporarily Skip MPI on p6-b200.48xlarge

MPI jobs intermittently takes longer and time out on p6-b200.48xlarge. This commit temporarily skip the test to let other checks in the test run

hanwen-cluster committed:
Merge branch 'develop' into developsep16
